web/modules/custom/joinup_core/joinup_core.post_update.php

```diff
@@ -1136,3 +1136,81 @@ function joinup_core_post_update_delete_orphaned_solutions() {
     $solution->delete();
   }
 }
+
+/**
+ * Fix all needed timestamps and created dates for content.
+ */
+function joinup_core_post_update_0001_set_migrated_content_timestamp(&$sandbox) {
+  $connection = \Drupal::database();
+
+  // Set the oldest revision timestamp to the created date for content.
+  // The created time was retained throughout the migration but the revision
+  // timestamp was set anew during the migration.
+  $query = <<<QUERY
+UPDATE {node_revision} nr
+INNER JOIN (
+    SELECT nid, MIN(created) AS created_timestamp
+    FROM node_field_revision
+    GROUP BY nid, vid
+    ORDER BY vid ASC
+  ) nfr
+  ON nr.nid = nfr.nid
+SET nr.revision_timestamp = nfr.created_timestamp
+WHERE nr.nid < 700000;
+QUERY;
+  $connection->query($query)->execute();
+
+  // Set the created timestamp all content revisions to the earliest revision
+  // timestamp of that node. This will undo our workaround change the created
+  // time upon initial publication.
+  $query = <<<QUERY
+UPDATE {node_field_revision} nfr
+ INNER JOIN (
+    SELECT nid, MIN(revision_timestamp) as revision_timestamp
+    FROM node_revision
+    GROUP BY nid
+) nr ON nfr.nid = nr.nid
+SET nfr.created = nr.revision_timestamp
+WHERE nfr.nid = nr.nid AND nfr.nid >= 700000
```

Comment:
So, we're joining on `nfr.nid = nr.nid` but the we also add this condition to WHERE?

Reply:
This might be a leftover because I did switch from INNER JOIN to raw tables select and back.